backend/app/controllers/application_controller.rb

```diff
@@ -3,7 +3,11 @@
 
 # application_controller
 class ApplicationController < ActionController::API
-  # Check whether a token has a specific permission_id (0 = anything that is not retired).
+  # Check whether a token has a specific permission_id.
+  # permission_id ==  0:                 anything         (not retired)
```

Comment:
Let's discuss.  
- I need a wildcard - I am using "0", we can use something else, but we would still have to define what that is...

- The system creates a hash of key-value pairs for permissions, { id: <value> } and would use that.

In my architecture, whenever the system sends permissions to the front-end it would send that hash, and in the case of selecting a permission, the front-end would presumably create something like this

<select>
<option value="<id>"><value></option>
...
</select>

or something like this

<input type=checkbox name="<id>"> <value>

and would pass the <id> back to the back-end for processing instead of passing the <value> back for processing.

In my experience that is more robust, offers faster performance, is easier to lock down, etc. etc. etc.  (You use one ".to_i", you do integer processing (which is faster than strings), and you and avoid all kinds of escaping to prevent hacking vulnerabilities, etc....)

Happy to discuss and do differently, though.